templates/an-cash-dao/contracts/BaseTemplate.sol

```diff
@@ -0,0 +1,359 @@
+    }
+
+    /* AGENT */
+
+    function _installDefaultAgentApp(Kernel _dao) internal returns (Agent) {
+        bytes memory initializeData = abi.encodeWithSelector(Agent(0).initialize.selector);
+        Agent agent = Agent(_installDefaultApp(_dao, AGENT_APP_ID, initializeData));
+        // We assume that installing the Agent app as a default app means the DAO should have its
+        // Vault replaced by the Agent. Thus, we also set the DAO's recovery app to the Agent.
+        _dao.setRecoveryVaultAppId(AGENT_APP_ID);
+        return agent;
+    }
+
+    function _installNonDefaultAgentApp(Kernel _dao) internal returns (Agent) {
+        bytes memory initializeData = abi.encodeWithSelector(Agent(0).initialize.selector);
+        return Agent(_installNonDefaultApp(_dao, AGENT_APP_ID, initializeData));
+    }
+
+    function _createAgentPermissions(ACL _acl, Agent _agent, address _grantee, address _manager) internal {
+        _acl.createPermission(_grantee, _agent, _agent.EXECUTE_ROLE(), _manager);
+        _acl.createPermission(_grantee, _agent, _agent.RUN_SCRIPT_ROLE(), _manager);
+    }
+
+    /* VAULT */
+
+    function _installVaultApp(Kernel _dao) internal returns (Vault) {
+        bytes memory initializeData = abi.encodeWithSelector(Vault(0).initialize.selector);
+        return Vault(_installDefaultApp(_dao, VAULT_APP_ID, initializeData));
+    }
+
+    function _createVaultPermissions(ACL _acl, Vault _vault, address _grantee, address _manager) internal {
+        _acl.createPermission(_grantee, _vault, _vault.TRANSFER_ROLE(), _manager);
+    }
+
+    /* DISPUTABLE VOTING */
+
+    function _installDisputableVotingApp(Kernel _dao, MiniMeToken _token, uint64[7] memory _votingSettings) internal returns (DisputableVoting) {
+        return _installDisputableVotingApp(_dao, _token, _votingSettings[0], _votingSettings[1], _votingSettings[2], _votingSettings[3], _votingSettings[4], _votingSettings[5], _votingSettings[6]);
+    }
+
+    function _installDisputableVotingApp(
+        Kernel _dao,
+        MiniMeToken _token,
+        uint64 _duration,
+        uint64 _support,
+        uint64 _acceptance,
+        uint64 _delegatedVotingPeriod,
+        uint64 _quietEndingPeriod,
+        uint64 _quietEndingExtension,
+        uint64 _executionDelay
+    )
+        internal returns (DisputableVoting)
+    {
+        bytes memory initializeData = abi.encodeWithSelector(DisputableVoting(0).initialize.selector, _token, _duration, _support, _acceptance, _delegatedVotingPeriod, _quietEndingPeriod, _quietEndingExtension, _executionDelay);
+        return DisputableVoting(_installNonDefaultApp(_dao, DISPUTABLE_VOTING_APP_ID, initializeData));
+    }
+
+    function _createDisputableVotingPermissions(
+        ACL _acl,
+        DisputableVoting _voting,
+        address _grantee,
+        address _manager
+    )
+        internal
+    {
+        _acl.createPermission(ANY_ENTITY, _voting, _voting.CHALLENGE_ROLE(), _manager);
```

Comment:
Maybe useful to attach another param as the challenger, but not a huge deal either way.